libraries/node-core-library/src/FileSystem.ts

```diff
@@ -521,4 +774,34 @@ export class FileSystem {
   public static getRealPath(linkPath: string): string {
     return fsx.realpathSync(linkPath);
   }
+
+  /**
+   * An async version of {@link FileSystem.getRealPath}.
+   */
+  public static getRealPathAsync(linkPath: string): Promise<string> {
+    return LegacyAdapters.convertCallbackToPromise(fsx.realpath, linkPath);
```

Comment:
@iclanton The type inference is not working correctly here.  For example, if I do this:
```ts
// inferred type of x is Promise<unknown>
let x = LegacyAdapters.convertCallbackToPromise(fsx.realpath, linkPath);
```
Which means that, for example, if I do this:
```ts
  public static getRealPathAsync(linkPath: string): Promise<string> {
    //return LegacyAdapters.convertCallbackToPromise(fsx.realpath, linkPath);
    return LegacyAdapters.convertCallbackToPromise(fsx.readFile, linkPath);  // readFile() returns Promise<Buffer>
  }
```
...then the mismatched return type is silently ignored.  The compiler does not detect it, because `unknown` gets silently converted to `Promise<Buffer>`.

I feel like we could:
- enable more of the **tsconfig.json** validation checks, which might catch this problem (the RSC default options are somewhat lax for modern compilers)
- improve the `LegacyCallback` declaration somehow so that type inference is performed correctly (maybe ask a type algebraist like @ThomasMichon for help heheh)
- always explicitly provide generic parameters for `convertCallbackToPromise()`, which is a simple solution

Reply:
Let's try to fix this in a later change. This is definitely a problem.